### PR TITLE
Fix assertSimilarMarkup() helper and tests that it obfuscated

### LINKED ISSUE #5420: Helper function assertSimilarMarkup() strips attributes instead of normalizing them
## Bug Description

The helper function `assertSimilarMarkup` in the `MarkupComparison` trait is supposed to normalize all of the HTML that would cause a strict string comparison to fail but not actually have an effect on how the HTML is interpreted, like the presence or absence of an attribute's empty value (`foo` <=> `foo=""`).

However, the code to normalize the attributes within each element is broken and actually fails to return the attributes. Therefore, only the element itself is compared, and attributes are just ignored even if completely different.

This has caused a few proper test failures to slip through that need to be fixed.

## Expected Behaviour

`assertSimilarMarkup()` should flag anything that is an actual difference in how HTML is interpreted.

## Additional context

 - WordPress version:
 - Plugin version:
 - Gutenberg plugin version (if applicable):
 - AMP plugin template mode:
 - PHP version:
 - OS:
 - Browser: [e.g. chrome, safari]

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by schlessera**
## Summary

This PR fixes the `assertSimilarMarkup()` helper function found in the `MarkupComparison` trait, as well as all the tests that it obfuscated that were already broken without us knowing it.
Fixes #5420

## Checklist

- [x] My pull request is addressing an open issue (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by schlessera** [comment]
Pushed the initial fix to record all the broken tests that were hidden as a result. See Travis results above ^^^

**Comment by schlessera** [comment]
There's 2 test failures now in `AMP_Tag_And_Attribute_Sanitizer_Test::test_sanitize()`, but those tests don't even use the `assertSimilarMarkup()` helper, but rather the `assertEqualMarkup()` instead. Did I break something when adding the comments there? The code looks correct to me. Maybe a pure coincidence and something else just broke at the same time?

**Comment by schlessera** [comment]
Oh, I checked the helper in the wrong location. I did indeed drop a statement from the main plugin's helper.

File: lib/optimizer/src/Transformer/ServerSideRendering.php [hunk 1]
```diff
@@ -953,7 +953,10 @@ private function extractSizesAttributeCss(Document $document, DOMElement $elemen
      */
     private function extractHeightsAttributeCss(Document $document, DOMElement $element, DOMAttr $attribute)
     {
-        $this->customSizerStyles[$document->getElementId($element)] = '';
+        // TODO: I'm not sure why I initially added this here, it looks very intentional.
+        // However, it doesn't match what the NodeJS version does, which is to add padding-top
+        // to the inline style of the element.
+        // $this->customSizerStyles[$document->getElementId($element)] = '';
```
Comment (schlessera) [inline lib/optimizer/src/Transformer/ServerSideRendering.php:959]: This change is what's needed to get in sync with the NodeJS version. However, the entire code construct for `$this->customSizerStyles` seems to have only been included for skipping the styles here, and I don't know why I added this in the first place, so I'm unsure about this "fix" and wonder whether the NodeJS version is correct in the first place.
I'll revisit this as soon as we have Percy visual regression tests, to diligently compare non-optimized and optimized versions of everything.

File: lib/optimizer/tests/Transformer/ServerSideRenderingTest.php [hunk 2]
```diff
@@ -220,15 +220,15 @@ public function dataTransform()
...
                 $expectWithoutBoilerplate(
-                    '<amp-img height="300" layout="responsive" srcset="https://acme.org/image1.png 320w, https://acme.org/image2.png 640w, https://acme.org/image3.png 1280w" sizes="(min-width: 200px) 200px, (min-width: 320px) 320px, 100vw" src="https://acme.org/image1.png" width="400" id="i-amp-0" class="i-amphtml-layout-responsive i-amphtml-layout-size-defined" i-amphtml-layout="responsive"><i-amphtml-sizer style="display:block;padding-top:75.0000%;"></i-amphtml-sizer></amp-img>',
+                    '<amp-img class="i-amphtml-layout-responsive i-amphtml-layout-size-defined" height="300" i-amphtml-layout="responsive" id="i-amp-0" layout="responsive" src="https://acme.org/image1.png" srcset="https://acme.org/image1.png 320w, https://acme.org/image2.png 640w, https://acme.org/image3.png 1280w" width="400"><i-amphtml-sizer style="display:block;padding-top:75.0000%;"></i-amphtml-sizer></amp-img>',
```
Comment (westonruter) [inline lib/optimizer/tests/Transformer/ServerSideRenderingTest.php:223]: The difference here is that after the `sizes="(min-width: 200px) 200px, (min-width: 320px) 320px, 100vw"` attribute is removed.

File: lib/optimizer/tests/Transformer/ServerSideRenderingTest.php [hunk 3]
```diff
@@ -220,15 +220,15 @@ public function dataTransform()
...
                 ),
                 [],
             ],
 
             'bad sizes attribute' => [
                 $input('<amp-img height="300" layout="responsive" srcset="https://acme.org/image1.png 320w, https://acme.org/image2.png 640w, https://acme.org/image3.png 1280w" sizes=",,," src="https://acme.org/image1.png" width="400"></amp-img>'),
-                $expectWithBoilerplate('<amp-img height="300" layout="responsive" srcset="https://acme.org/image1.png 320w, https://acme.org/image2.png 640w, https://acme.org/image3.png 1280w" sizes=",,,"  src="https://acme.org/image1.png" width="400"><i-amphtml-sizer style="display:block;padding-top:75.0000%;"></i-amphtml-sizer></amp-img>'),
+                $expectWithBoilerplate('<amp-img class="i-amphtml-layout-responsive i-amphtml-layout-size-defined" height="300" i-amphtml-layout="responsive" layout="responsive" sizes=",,," src="https://acme.org/image1.png" srcset="https://acme.org/image1.png 320w, https://acme.org/image2.png 640w, https://acme.org/image3.png 1280w" width="400"><i-amphtml-sizer style="display:block;padding-top:75.0000%;"></i-amphtml-sizer></amp-img>'),
```
Comment (westonruter) [inline lib/optimizer/tests/Transformer/ServerSideRenderingTest.php:231]: Difference here is that these attributes are added:

* `class="i-amphtml-layout-responsive i-amphtml-layout-size-defined"`
* `i-amphtml-layout="responsive"`

File: lib/optimizer/tests/Transformer/ServerSideRenderingTest.php [hunk 4]
```diff
@@ -265,7 +265,7 @@ public function dataTransform()
             'bad heights attribute' => [
                 $input('<amp-img height="256" heights=",,," layout="responsive" width="320"></amp-img>'),
                 // This adds an ID as it stores the CSS to inline before the actual error is detected.
-                $expectWithBoilerplate('<amp-img height="256" heights=",,," layout="responsive" width="320" id="i-amp-0"><i-amphtml-sizer style="display:block;padding-top:75.0000%;"></i-amphtml-sizer></amp-img>'),
+                $expectWithBoilerplate('<amp-img class="i-amphtml-layout-responsive i-amphtml-layout-size-defined" height="256" heights=",,," i-amphtml-layout="responsive" layout="responsive" width="320"><i-amphtml-sizer style="display:block;padding-top:80.0000%;"></i-amphtml-sizer></amp-img>'),
```
Comment (westonruter) [inline lib/optimizer/tests/Transformer/ServerSideRenderingTest.php:268]: Difference is `id` is removed, while these are added:

* `class="i-amphtml-layout-responsive i-amphtml-layout-size-defined"`
* `i-amphtml-layout="responsive"`

And the `padding-top` changes from 75% to 80%.

File: lib/optimizer/tests/Transformer/ServerSideRenderingTest.php (hunk 2 above)
Comment (westonruter) [reply lib/optimizer/tests/Transformer/ServerSideRenderingTest.php:223]: Node.js Optimizer is generating:

```html
<amp-img height="300" layout="responsive" srcset="https://acme.org/image1.png 320w, https://acme.org/image2.png 640w, https://acme.org/image3.png 1280w" src="https://acme.org/image1.png" width="400" class="i-amphtml-layout-responsive i-amphtml-layout-size-defined" i-amphtml-layout="responsive" id="i-amp-0"><i-amphtml-sizer style="display:block;padding-top:75.0000%;"></i-amphtml-sizer></amp-img>
```

Which matches!

File: lib/optimizer/tests/Transformer/ServerSideRenderingTest.php (hunk 3 above)
Comment (westonruter) [reply lib/optimizer/tests/Transformer/ServerSideRenderingTest.php:231]: Matches Node.js output.

File: lib/optimizer/tests/Transformer/ServerSideRenderingTest.php (hunk 4 above)
Comment (westonruter) [reply lib/optimizer/tests/Transformer/ServerSideRenderingTest.php:268]: `256 / 320 = 0.80`

Comment (westonruter) [reply lib/optimizer/tests/Transformer/ServerSideRenderingTest.php:268]: Matches output with Node.js Optimizer.

**Review by westonruter — APPROVED** [review]

**Pull request merged**